pkg/index/cache/cache.go

```diff
@@ -0,0 +1,315 @@
+// Copyright 2023 The Falco Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/falcosecurity/falcoctl/internal/config"
+	"github.com/falcosecurity/falcoctl/internal/consts"
+	"github.com/falcosecurity/falcoctl/pkg/index"
+)
+
+// Cache manages the index files.
+type Cache struct {
+	*index.MergedIndexes
+	localIndexes     *index.Config
+	localIndexesFile string
+	indexesDir       string
+	// Track the new indexes that need to be saved locally when writing the cache to file.
+	fetchedIndexes []*index.Index
+	// Track the indexes that have been removed, needed when writing the cache to file.
+	removedIndexes []string
+}
+
+// New creates a new cache object. For each entry in the indexes.yaml file it loads the respective index file
+// found on the disk or fetches it if not found. If there is an entry in the indexes.yaml file but its index file does not exist on the disk
+// then it will error.
+func New(ctx context.Context, indexFile, indexesDir string) (*Cache, error) {
```

Comment:
Although it looks good to me, the documentation does not describe what an `indexFile` is.
In this context, I assume an _`indexFile` is the index file of the artifact's indexes files contained into `indexesDir`_. Is that correct?

Reply:
That's correrct. We did not document it, yet. Will do in the future, since it is an internal configuration and not a user facing one.